cpp/src/io/orc/writer_impl.cu

```diff
@@ -631,6 +651,14 @@ rmm::device_buffer writer::impl::encode_columns(orc_column_view *columns,
     }
   }
 
+  for (auto &cnt_in : validity_check_inputs) {
+    auto const valid_counts = segmented_count_set_bits(cnt_in.second.mask, cnt_in.second.indices);
+    CUDF_EXPECTS(std::none_of(valid_counts.cbegin(),
+                              valid_counts.cend(),
+                              [](auto valid_count) { return valid_count % 8; }),
+                 "Boolean column can't be encoded correctly. Please convert to int8.");
```

Comment:
I would update this to point to the github issue and make it something along the lines of:

```
There's currently a bug in encoding boolean columns. Suggested workaround is to convert to int8 type. Please see https://github.com/rapidsai/cudf/issues/1234 for more information.
```